src/test/java/org/broadinstitute/hellbender/tools/walkers/CombineGVCFsIntegrationTest.java

```diff
@@ -0,0 +1,379 @@
+        }
+        Files.copy(gatk3Result.toPath(), expectedResultsDir.resolve(outputFile.getName()), StandardCopyOption.REPLACE_EXISTING);
+
+        assertVariantContextsMatch(Arrays.asList(inputs), gatk3Result, extraArgs, reference);
+    }
+
+    @Test(dataProvider = "gvcfsToGenotype")
+    public void compareToGATK3ExpectedResults(File[] inputs, File outputFile, List<String> extraArgs, String reference) throws IOException, NoSuchAlgorithmException {
+        assertVariantContextsMatch(Arrays.asList(inputs), outputFile, extraArgs, reference);
+    }
+
+    protected static void runProcess(ProcessController processController, String[] command) {
+        final ProcessSettings prs = new ProcessSettings(command);
+        prs.getStderrSettings().printStandard(true);
+        prs.getStdoutSettings().printStandard(true);
+        final ProcessOutput output = processController.exec(prs);
+        Assert.assertEquals(output.getExitValue(), 0, "Process exited with non-zero value. Command: "+ Arrays.toString(command) + "\n");
+    }
+
+
+
+    private void assertVariantContextsMatch(List<File> inputs, File expected, List<String> extraArgs, String reference) throws IOException {
+        final VCFHeader header = getHeaderFromFile(expected);
+
+        runCombineGVCFSandAssertSomething(inputs, expected, extraArgs, (a, e) -> {
+            VariantContextTestUtils.assertVariantContextsAreEqualAlleleOrderIndependent(a, e, Arrays.asList(), header);
+        }, reference);
+    }
+
+    private void runCombineGVCFSandAssertSomething(List<File> inputs, File expected, List<String> additionalArguments, BiConsumer<VariantContext, VariantContext> assertion, String reference) throws IOException {
+        final File output = createTempFile("genotypegvcf", ".vcf");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addReference(new File(reference))
+                .addOutput(output);
+        for (File input: inputs) {
+            args.addArgument("V", input.getAbsolutePath());
+        }
+
+        // Handling a difference in syntax between GATK3 and GATK4 wrt. annotation groups
+        additionalArguments = additionalArguments.stream().map(a -> a.contains("Standard") ? a + "Annotation" : a).collect(Collectors.toList());
+        additionalArguments.forEach(args::add);
+
+        Utils.resetRandomGenerator();
+        runCommandLine(args);
+
+        final List<VariantContext> expectedVC = getVariantContexts(expected);
+        final List<VariantContext> actualVC = getVariantContexts(output);
+        assertForEachElementInLists(actualVC, expectedVC, assertion);
+    }
+
+    /**
+     * Returns a list of VariantContext records from a VCF file
+     *
+     * @param vcfFile VCF file
+     * @return list of VariantContext records
+     * @throws IOException if the file does not exist or can not be opened
+     */
+    private static List<VariantContext> getVariantContexts(final File vcfFile) throws IOException {
```

Comment:
ok, I don't have any idea what I was thinking when I wrote this.... could you replace this monstrosity with

```
    private static List<VariantContext> getVariantContexts(final File vcfFile) throws IOException {
        try(final FeatureDataSource<VariantContext> variantContextFeatureDataSource = new FeatureDataSource<>(vcfFile)) {
            return IteratorUtils.toList(variantContextFeatureDataSource.iterator());
        }
    }
```

same in the source.... method, which maybe we should move to some common place?

Reply:
yes fixed in both places. Much nicer and autoclosing too.